pybamm/solvers/base_solver.py

```diff
@@ -640,73 +663,98 @@ def solve(self, model, t_eval=None, external_variables=None, inputs=None):
         # object, restarting the solver at each discontinuity (and recalculating a
         # consistent state afterwards if a dae)
         old_y0 = model.y0
-        solution = None
+        solutions = None
         for start_index, end_index in zip(start_indices, end_indices):
             pybamm.logger.info(
                 "Calling solver for {} < t < {}".format(
                     t_eval_dimensionless[start_index] * model.timescale_eval,
                     t_eval_dimensionless[end_index - 1] * model.timescale_eval,
                 )
             )
-            new_solution = self._integrate(
-                model, t_eval_dimensionless[start_index:end_index], ext_and_inputs
-            )
-            new_solution.solve_time = timer.time()
-            if solution is None:
-                solution = new_solution
+            ninputs = len(ext_and_inputs_list)
+            if ninputs == 1:
+                new_solution = self._integrate(
+                    model,
+                    t_eval_dimensionless[start_index:end_index],
+                    ext_and_inputs_list[0],
+                )
+                new_solutions = [new_solution]
             else:
-                solution.append(new_solution, start_index=0)
+                with mp.Pool(processes=nproc) as p:
+                    new_solutions = p.starmap(
+                        self._integrate,
+                        zip(
+                            [model] * ninputs,
+                            [t_eval_dimensionless[start_index:end_index]] * ninputs,
+                            ext_and_inputs_list,
+                        ),
+                    )
+            # Setting the solve time for each segment
+            # Not sure why required here since overall solve time
+            # is set further down
+            solve_time = timer.time()
+            for sol in new_solutions:
+                sol.solve_time = solve_time
+            if start_index == start_indices[0]:
+                solutions = [sol for sol in new_solutions]
+            else:
+                for i, new_solution in enumerate(new_solutions):
+                    solutions[i].append(new_solution, start_index=0)
 
-            if solution.termination != "final time":
+            if solutions[0].termination != "final time":
                 break
 
             if end_index != len(t_eval_dimensionless):
```

Comment:
Again, the content of this `if` statement assumes that, if there are discontinuities, the `inputs` was a single dictionary, and therefore we're fine with just using `solutions[0]` and `ext_and_inputs_list[0]` to compute the restart state.